### ISSUE TITLE
EPERM on Windows when running in parallel

### ISSUE THREAD

**Issue by SimenB** [npm/write-file-atomic#10]
Running `del` in parallel on our CI running on Windows, throws EPERM. I was told to create an issue here.

Ref yeoman/update-notifier#68

Our stacktrace:
```
error   31-Mar-2016 14:15:50    Error: EPERM: operation not permitted, rename 'C:\Windows\system32\config\systemprofile\.config\configstore\update-notifier-del-cli.json.1874881583' -> 'C:\Windows\system32\config\systemprofile\.config\configstore\update-notifier-del-cli.json'
error   31-Mar-2016 14:15:50        at Error (native)
error   31-Mar-2016 14:15:50        at Object.fs.renameSync (fs.js:681:18)
error   31-Mar-2016 14:15:50        at Function.writeFileSync [as sync] (D:\Bamboo\bamboo-home\xml-data\build-dir\SON-FRON-JOB1\code\node_modules\write-file-atomic\index.js:39:8)
error   31-Mar-2016 14:15:50        at Object.create.all.set (D:\Bamboo\bamboo-home\xml-data\build-dir\SON-FRON-JOB1\code\node_modules\configstore\index.js:63:21)
error   31-Mar-2016 14:15:50        at Object.Configstore (D:\Bamboo\bamboo-home\xml-data\build-dir\SON-FRON-JOB1\code\node_modules\configstore\index.js:28:11)
error   31-Mar-2016 14:15:50        at new UpdateNotifier (D:\Bamboo\bamboo-home\xml-data\build-dir\SON-FRON-JOB1\code\node_modules\update-notifier\index.js:34:17)
error   31-Mar-2016 14:15:50        at module.exports (D:\Bamboo\bamboo-home\xml-data\build-dir\SON-FRON-JOB1\code\node_modules\update-notifier\index.js:108:23)
error   31-Mar-2016 14:15:50        at Object.<anonymous> (D:\Bamboo\bamboo-home\xml-data\build-dir\SON-FRON-JOB1\code\node_modules\del-cli\cli.js:30:1)
error   31-Mar-2016 14:15:50        at Module._compile (module.js:435:26)
error   31-Mar-2016 14:15:50        at Object.Module._extensions..js (module.js:442:10)
```

**Comment by dizlexik** [comment]
I'm having the same issue intermittently with AVA test runner when I run multiple test processes in parallel during my build. AVA uses `update-notifier` which uses `configstore` which uses `write-file-atomic`.

**Comment by larsthorup** [comment]
I also see this issue quite frequently with nodemon where I run multiple nodemon commands in parallel (for watching several micro services). nodemon uses update-notifier, etc.

**Comment by SimenB** [comment]
Should be fixed by #13

**Comment by ma-zal** [comment]
Are you sure, that problem has been fixed? 

I see the error ```Error: EPERM: operation not permitted, rename '...uFmzrijJmV3QUA1pWUZnUe4Si9fYJVWx.json.1817276499' -> '...uFmzrijJmV3QUA1pWUZnUe4Si9fYJVWx.json'``` when parallel access to API from one client is performed.

I'm using [passport](https://github.com/jaredhanson/passport) with [session-file-store](https://github.com/valery-barysok/session-file-store), that uses [write-file-atomic 1.2.0](https://github.com/npm/write-file-atomic).
